### DISCUSSION TITLE
The return type of the Prisma client method is inferred as any

### DISCUSSION THREAD

**Discussion by worjs** [prisma/prisma#20041] [Q&A]
Hi, I'm using prisma in my nestjs project.
During the project, I used the `findMany` function to query a model called `union`. However, I noticed that the return type is inferred as `any`. When I tried with other models, the return type was appropriately inferred, but only the `union`model's type was not worked. I also checked the types in the Prisma-generated types files, and the `union` type seems to be well-defined, with related client methods properly defined as well. However, only the return type is not inferred correctly. The `prisma generate` command also works fine. Below is a screenshot showing the schema and the inference of the return type as `any` for the union model. Is there something I might have missed?

```prisma
model Union {
  id           String    @id @default(uuid())
  name         String
  accountId    String    @unique @map("account_id")
  password     String
  photo        String?
  universityId Int?      @unique @map("university_id")
  schoolId     Int?      @unique @map("school_id")
  departmentId Int?      @unique @map("department_id")
  createdAt    DateTime  @default(now()) @map("created_at")
  updatedAt    DateTime  @updatedAt @map("updated_at")
  deletedAt    DateTime? @map("deleted_at")

  university University? @relation(fields: [universityId], references: [id])
  school     School?     @relation(fields: [schoolId], references: [id])
  department Department? @relation(fields: [departmentId], references: [id])

  userUnion UserUnion[]
  notice    Notice[]

  @@map("union")
}
```
<img width="602" alt="스크린샷 2023-07-03 오후 8 05 38" src="https://github.com/prisma/prisma/assets/20869611/9891dd06-394b-4c4b-9723-8bc74ffb50a7">

**Comment by ludralph** [comment]
HI @worjs 👋 

You can solve this by using the Prisma Validator [API](https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/prisma-validator). You can take a look at this Stackoverflow [answer](https://stackoverflow.com/a/68377797/1645620) for a detailed step on how to solve this same issue.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by worjs** [reply]
That's not the problem. I understood Prisma validator as a function to ensure the type-safe of query statements. And applying it does not solve the problem of queries such as prisma.findMany returning any type.
I personally think that the word "union" is already being used for other purposes in Prisma..

**Reply by ludralph** [reply]
Hi @worjs, You are correct that the word `Union` used as your model name is the cause. There is an open issue that documents this.

- https://github.com/prisma/prisma/issues/12469#:~:text=UnEnumerate-,Union,-IntersectOf

 You can also see a list of reserved names in Prisma [here](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#naming-conventions)

If this answers your question, it would be great if you could mark this Discussion as answered to indicate that it has been resolved.

Otherwise, please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by ludralph** [reply]
Hi @worjs ,

Do you have any more questions around this issue? If no, it would be really helpful if you could quickly **mark this Discussion as _answered_** to indicate that your issue has been resolved. 

In case you do have any outstanding questions, feel free to let us know and we'd be happy to help!

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem 🙏
